src/test/java/email/generator/hm/FileSourceReaderTest.java

```diff
@@ -0,0 +1,28 @@
+package email.generator.hm;
+
+
+import email.generator.hm.exception.ArgumentMissingForMailGeneratorException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvFileSource;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class FileSourceReaderTest {
+    @ParameterizedTest
+    @CsvFileSource(resources = "/template.csv")
```

Comment:
File reading should be implemented in the application not in tests.
The idea was to mock file reading in tests